main.py

```
@@ -15,6 +20,24 @@ class Plugin:
     lock_script_path = os.path.join(decky.DECKY_PLUGIN_DIR, "assets", "lock_wifi.sh")
     unlock_script_path = os.path.join(decky.DECKY_PLUGIN_DIR, "assets", "unlock_wifi.sh")
     state_file_path = os.path.join(decky.DECKY_PLUGIN_RUNTIME_DIR, "wifi_lock_state.json")
+    settings_file_path: Path = Path(decky.DECKY_PLUGIN_SETTINGS_DIR) / "wifi_locker_settings.json"
+
+    def _load_settings(self) -> PluginSettings:
+        """Load plugin settings from disk, returning defaults if absent or malformed."""
+        try:
+            if self.settings_file_path.exists():
+                return json.loads(self.settings_file_path.read_text())
+        except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)